packages/flutter_driver/lib/src/driver/timeline_summary.dart

```diff
@@ -84,11 +90,28 @@ class TimelineSummary {
   /// The total number of frames recorded in the timeline.
   int countFrames() => _extractFrameDurations().length;
 
+  /// The total number of rasterizer cycles recorded in the timeline.
+  int countRasterizations() => _extractGpuRasterizerDrawDurations().length;
+
   /// Encodes this summary as JSON.
   Map<String, dynamic> get summaryJson {
     final SceneDisplayLagSummarizer sceneDisplayLagSummarizer = _sceneDisplayLagSummarizer();
 
     return <String, dynamic>{
+      'frame_build_times': _extractFrameDurations()
+          .map<int>((Duration duration) => duration.inMicroseconds)
+          .toList(),
+      'frame_rasterizer_times': _extractGpuRasterizerDrawDurations()
+          .map<int>((Duration duration) => duration.inMicroseconds)
+          .toList(),
+      'frame_begin_times': _extractBeginTimestamps(kBuildFrameEventName)
+          .map<int>((Duration duration) => duration.inMicroseconds)
+          .toList(),
+      'frame_rasterizer_begin_times': _extractBeginTimestamps(kRasterizeFrameEventName)
+          .map<int>((Duration duration) => duration.inMicroseconds)
+          .toList(),
+      'frame_count': countFrames(),
+      'frame_rasterizer_count': countRasterizations(),
```

Comment:
The addition of `frame_rasterizer_begin_times` looks good.

For the order change, considering that most text editors will show the top of a file by default, is it better to still keep the key statistics on top? Or maybe we should only write the key statistics (e.g., average, percentiles) in the summary json file, and write all timestamps in another json file?

Reply:
Since the output at the end of the run is only loosely related to what is in the file, we could have the output put important information at the end, but the file contain important information at the top. Let me look into that.

Reply:
Actually, I'll make the changes separately and undo the ordering in this PR.